conjure-java-core/src/integrationInput/java/test/prefix/com/palantir/product/PrimitiveOptionalsExample.java

```diff
@@ -263,25 +263,26 @@ private boolean equalTo(PrimitiveOptionalsExample other) {
     public int hashCode() {
         int result = memoizedHashCode;
         if (result == 0) {
-            result = Objects.hash(
-                    this.num,
-                    this.bool,
-                    this.integer,
-                    this.safelong,
-                    this.rid,
-                    this.bearertoken,
-                    this.uuid,
-                    this.map,
-                    this.list,
-                    this.set,
-                    this.aliasOne,
-                    this.aliasTwo,
-                    this.aliasList,
-                    this.aliasMap,
-                    this.aliasOptional,
-                    this.aliasOptionalMap,
-                    this.aliasOptionalList,
-                    this.aliasOptionalSet);
+            int hash = 4441;
+            hash += (hash << 5) + Objects.hashCode(this.num);
+            hash += (hash << 5) + Objects.hashCode(this.bool);
+            hash += (hash << 5) + Objects.hashCode(this.integer);
+            hash += (hash << 5) + Objects.hashCode(this.safelong);
+            hash += (hash << 5) + Objects.hashCode(this.rid);
+            hash += (hash << 5) + Objects.hashCode(this.bearertoken);
+            hash += (hash << 5) + Objects.hashCode(this.uuid);
+            hash += (hash << 5) + Objects.hashCode(this.map);
+            hash += (hash << 5) + Objects.hashCode(this.list);
+            hash += (hash << 5) + Objects.hashCode(this.set);
+            hash += (hash << 5) + Objects.hashCode(this.aliasOne);
+            hash += (hash << 5) + Objects.hashCode(this.aliasTwo);
+            hash += (hash << 5) + Objects.hashCode(this.aliasList);
+            hash += (hash << 5) + Objects.hashCode(this.aliasMap);
+            hash += (hash << 5) + Objects.hashCode(this.aliasOptional);
+            hash += (hash << 5) + Objects.hashCode(this.aliasOptionalMap);
+            hash += (hash << 5) + Objects.hashCode(this.aliasOptionalList);
+            hash += (hash << 5) + Objects.hashCode(this.aliasOptionalSet);
```

Comment:
I'm curious about this hashing algorithm, normally we use something like
```java
hash = 31 * hash + fieldHashCode;
```
Which the jvm will optimize into
```java
hash = (hash << 5) - hash + fieldHashCode;
```

Reply:
What do you think about retaining the existing hashcode values so we can measure the impact of this change in isolation?

Reply:
I wonder if we should split out a separate private method to compute the hashcode, keeping this method small

Reply:
I can adjust the hash algorithm. I had initially modeled it after immutables, but keeping consistency with `Objects.hash(Objects...)` & `Arrays.hashCode(Object[])` makes sense.

Reply:
I don't have a strong preference on separate `generateHashCode()` method vs inline as I originally had former then inlined it in https://github.com/palantir/conjure-java/commit/8714f3031f7623e06349152e7ee190e457c3da61

Reply:
> I don't have a strong preference on separate generateHashCode() method vs inline as I originally had former then inlined it

inline is simpler code, happy to go with that

Reply:
updated